builder/vmware/common/hw_config.go

```diff
@@ -70,57 +79,69 @@ type HWConfig struct {
 	//
-	//   * `yield` (bool)      - This is an optional boolean that specifies
-	//     whether the vm should yield the cpu when polling the port. By
+	//   * `yield` (bool) - This is an optional boolean that specifies whether
+	//     the virtual machine should yield the CPU when polling the port. By
 	//     default, the builder will assume this as `FALSE`.
 	//
 	// * `AUTO:(yield)` - Specifies to use auto-detection to determine the
-	//   serial port to use. This has one option to determine how the VM should
-	//   support the serial port.
+	//   serial port to use. This has one option to determine how the virtual
+	//   machine should support the serial port.
 	//
-	//   * `yield` (bool) - This is an optional boolean that specifies
-	//     whether the vm should yield the cpu when polling the port. By
+	//   * `yield` (bool) - This is an optional boolean that specifies whether
+	//     the virtual machine should yield the CPU when polling the port. By
 	//     default, the builder will assume this as `FALSE`.
 	//
 	// * `NONE` - Specifies to not use a serial port. (default)
-	//
 	Serial string `mapstructure:"serial" required:"false"`
-	// This specifies a parallel port to add to the VM. It has the format of
-	// `Type:option1,option2,...`. Type can be one of the following values:
+	// Add a parallel port to add to the virtual machine. Use a format of
+	// `Type:option1,option2,...`. Allowed values for the field `Type` include:
 	// `FILE`, `DEVICE`, `AUTO`, or `NONE`.
 	//
-	// * `FILE:path` 		- Specifies the path to the local file to be used
-	//   for the parallel port.
+	// * `FILE:path` - Specifies the path to the local file to be used for the
+	//    parallel port.
 	//
-	// * `DEVICE:path`	 	- Specifies the path to the local device to be used
-	//   for the parallel port.
+	// * `DEVICE:path` - Specifies the path to the local device to be used for
+	//    the parallel port.
 	//
-	// * `AUTO:direction`   - Specifies to use auto-detection to determine the
+	// * `AUTO:direction` - Specifies to use auto-detection to determine the
 	//   parallel port. Direction can be `BI` to specify bidirectional
 	//   communication or `UNI` to specify unidirectional communication.
 	//
-	// * `NONE` 			- Specifies to not use a parallel port. (default)
+	// * `NONE` - Specifies to not use a parallel port. (default)
 	Parallel string `mapstructure:"parallel" required:"false"`
 }
 
 func (c *HWConfig) Prepare(ctx *interpolate.Context) []error {
 	var errs []error
 
-	if c.CpuCount < 0 {
-		errs = append(errs, fmt.Errorf("invalid number of cpus specified (cpus < 0): %d", c.CpuCount))
+	if c.Firmware != "" {
+		isValidFirmware := false
+		for _, firmware := range allowedFirmwareTypes {
+			if c.Firmware == firmware {
+				isValidFirmware = true
+				break
+			}
+		}
+		if !isValidFirmware {
+			errs = append(errs, fmt.Errorf("invalid 'firmware' type specified: %s; must be one of %s", c.Firmware, strings.Join(allowedFirmwareTypes, ", ")))
+		}
```

Comment:
Since you are just validating that the provided firmware value is valid you can use the std-lib to do the work for you here. 

```suggestion
	if (c.Firmware != "") && (slices.Contains(allowedFirmwareTypes, c.Firmware) == false) {
			errs = append(errs, fmt.Errorf("invalid 'firmware' type specified: %s; must be one of %s", c.Firmware, strings.Join(allowedFirmwareTypes, ", ")))
```

Reply:
Resolved in https://github.com/hashicorp/packer-plugin-vmware/commit/45c037f1538cda663def623f50ee7862c778bd70.